File: lista-de-exercicios/exercicio-7/Fibonacci.test.ts
import { jest } from '@jest/globals';
import { Fibonacci } from './Fibonacci';

describe('Classe Fibonacci', () => {
    test('Deve retornar um array com a série de Fibonacci do tamanho do número informado +1', () => {
        const fibonacci0 = new Fibonacci(0);
        const fibonacci1 = new Fibonacci(1);
        const fibonacci2 = new Fibonacci(2);
        const fibonacci3 = new Fibonacci(3);
        
        expect(fibonacci0.serializar()).toStrictEqual([0]);
        expect(fibonacci1.serializar()).toStrictEqual([0, 1]);
        expect(fibonacci2.serializar()).toStrictEqual([0, 1, 1]);
        expect(fibonacci3.serializar()).toStrictEqual([0, 1, 1, 2]);
    })

    test('Deve retornar um erro se entrar um número decimal ou negativo', () => {
        const fibonacciNegativo = new Fibonacci(-1);
        const fibonacciDecimal = new Fibonacci(1.2);
        const fibonacciNaN = new Fibonacci(NaN);

        expect(() => fibonacciNegativo.serializar()).toThrowError();
        expect(() => fibonacciDecimal.serializar()).toThrowError();
        expect(() => fibonacciNaN.serializar()).toThrowError();
    })
})

File: lista-de-exercicios-java/src/test/java/org/example/exercicio07/FibonacciTest.java
package org.example.exercicio07;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;

import static org.junit.jupiter.api.Assertions.*;

class FibonacciTest {

    @DisplayName("Deve retornar um array com a série de Fibonacci do tamanho do número informado +1")
    @Test
    void deveRetornarArrayDoTamanhoMais1() {
        Fibonacci fibo = new Fibonacci();
        ArrayList<Integer> fiboArray = fibo.serializar(1);

        assertEquals(2, fiboArray.size());
    }


    @DisplayName("Deve retornar o array correto")
    @Test
    void deveRetornarArrayCorreto() {
        Fibonacci fibo = new Fibonacci();
        ArrayList<Integer> fiboArray = fibo.serializar(9);

        assertArrayEquals(new Integer[] {0, 1, 1, 2, 3, 5, 8, 13, 21, 34}, fibo.serializar(9).toArray());
    }

    @DisplayName("Deve retornar um erro se entrar um número negativo")
    @Test
    void deveRetornarErroComNumeroNegativo() {
        Fibonacci fibo = new Fibonacci();

        assertThrows(IllegalArgumentException.class, () -> {
            fibo.serializar(-2);
        });
    }
}